lib/src/simple_op_store.rs

```diff
@@ -176,190 +353,186 @@ fn operation_metadata_from_proto(
-        if let Some(local_target) = &target.local_target {
-            branch_proto.local_target = MessageField::some(ref_target_to_proto(local_target));
-        }
+        let local_target = target.local_target.as_ref().map(ref_target_to_thrift);
+        let mut remote_branches = vec![];
         for (remote_name, target) in &target.remote_targets {
-            let mut remote_branch_proto = crate::protos::op_store::RemoteBranch::new();
-            remote_branch_proto.remote_name = remote_name.clone();
-            remote_branch_proto.target = MessageField::some(ref_target_to_proto(target));
-            branch_proto.remote_branches.push(remote_branch_proto);
+            remote_branches.push(simple_op_store_model::RemoteBranch::new(
+                remote_name.clone(),
+                ref_target_to_thrift(target),
+            ));
         }
-        proto.branches.push(branch_proto);
+        branches.push(simple_op_store_model::Branch::new(
+            name.clone(),
+            local_target,
+            remote_branches,
+        ));
     }
 
+    let mut tags = vec![];
     for (name, target) in &view.tags {
-        let mut tag_proto = crate::protos::op_store::Tag::new();
-        tag_proto.name = name.clone();
-        tag_proto.target = MessageField::some(ref_target_to_proto(target));
-        proto.tags.push(tag_proto);
+        tags.push(simple_op_store_model::Tag::new(
+            name.clone(),
+            ref_target_to_thrift(target),
+        ));
     }
 
+    let mut git_refs = vec![];
     for (git_ref_name, target) in &view.git_refs {
-        let mut git_ref_proto = crate::protos::op_store::GitRef::new();
-        git_ref_proto.name = git_ref_name.clone();
-        git_ref_proto.target = MessageField::some(ref_target_to_proto(target));
-        proto.git_refs.push(git_ref_proto);
+        git_refs.push(simple_op_store_model::GitRef::new(
+            git_ref_name.clone(),
+            ref_target_to_thrift(target),
+        ));
     }
 
-    if let Some(git_head) = &view.git_head {
-        proto.git_head = git_head.to_bytes();
-    }
+    let git_head = view.git_head.as_ref().map(|git_head| git_head.to_bytes());
 
-    proto
+    simple_op_store_model::View::new(
+        head_ids,
+        public_head_ids,
+        wc_commit_ids,
+        branches,
+        tags,
+        git_refs,
+        git_head,
+    )
 }
 
-fn view_from_proto(proto: &crate::protos::op_store::View) -> View {
+fn view_from_thrift(thrift: &simple_op_store_model::View) -> View {
     let mut view = View::default();
-    // For compatibility with old repos before we had support for multiple working
-    // copies
-    if !proto.wc_commit_id.is_empty() {
-        view.wc_commit_ids.insert(
-            WorkspaceId::default(),
-            CommitId::new(proto.wc_commit_id.clone()),
-        );
-    }
-    for (workspace_id, commit_id) in &proto.wc_commit_ids {
+    for (workspace_id, commit_id) in thrift.wc_commit_ids.iter() {
```

Comment:
nit: Is there no `IntoIter` impl for `&WhateverThriftThing`? I wonder if upstream has an open issue for that.